lib/portage/tests/util/test_install_mask.py

```diff
@@ -166,3 +166,29 @@ def testTrailingSlash(self):
                         install_mask_str, path
                     ),
                 )
+
+    def testSymlinkDir(self):
+        """
+        Test that masked symlinks to directories are removed.
+        """
+        tmp_dir = tempfile.mkdtemp()
+
+        try:
+            base_dir = os.path.join(tmp_dir, "foo")
+            target_dir = os.path.join(tmp_dir, "foo", "bar")
+            link_name = os.path.join(tmp_dir, "foo", "baz")
+
+            os.mkdir(base_dir)
+            os.mkdir(target_dir)
+            os.symlink(target_dir, link_name)
+
+            install_mask = InstallMask("/foo/")
+            install_mask_dir(tmp_dir, install_mask)
+            self.assertFalse(
+                os.path.lexists(link_name), "failed to remove {}".format(link_name)
+            )
+            self.assertFalse(
+                os.path.lexists(base_dir), "failed to remove {}".format(base_dir)
+            )
+        finally:
+            shutil.rmtree(tmp_dir)
```

Comment:
Looks like this file is missing imports for tempfile, shutil, and the install_mask_dir function.

Reply:
sorry, yes! Had to apply manually due to black. I realised but internet cut out, fixing now